indico/modules/events/registration/notifications.py

```diff
@@ -39,12 +39,15 @@ def _notify_registration(registration, template, to_managers=False, attach_rejec
             registration.state == RegistrationState.complete):
         attachments = get_ticket_attachments(registration)
 
+    template_name = template
     template = get_template_module('events/registration/emails/{}'.format(template),
                                    registration=registration, attach_rejection_reason=attach_rejection_reason)
     to_list = registration.email if not to_managers else registration.registration_form.manager_notification_recipients
     from_address = registration.registration_form.sender_address if not to_managers else None
     mail = make_email(to_list=to_list, template=template, html=True, from_address=from_address, attachments=attachments)
     user = session.user if session else None
+    mail = signals.core.before_notification_send.send('notify-registration', mail=mail, registration=registration,
```

Comment:
Did you test this? The return value of `.send()` is not what you may expect - remember, you can have more than one handler for a signal... Probably better to not use it at all and document that `mail` should be mutated if necessary.

Reply:
> Probably better to not use it at all and document that mail should be mutated if necessary.

👍